### PR TITLE
Eliminate start methods

### PR DESCRIPTION
**Pull request by dpw**
Most start methods in the router are trivially redundant.  Apart from the redundancy, start methods raise the question of the state of an object prior to the start method being called (i.e. which fields are valid, which methods can be called, etc.).  So they should only be used when there is a reason.

(Extracted from #991)

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -84,6 +84,9 @@ func NewRouter(config Config, name PeerName, nickName string) *Router {
 	return router
 }
 
+// Start listening for packets from containers, TCP connections and
```
Comment (rade) [inline router/router.go]: Shouldn't really mention 'containers' here; the router is largely unaware of them.

File: router/gossip_test.go [hunk 2]
```diff
@@ -22,11 +22,8 @@ type mockChannelConnection struct {
 // for Routes and GossipSenders.
 func NewTestRouter(name string) *Router {
```
Comment (rade) [inline router/gossip_test.go]: The comment above this needs updating since we now certainly do have a bunch of goroutines.

**Comment by rade** [comment]
rest looks fine.

File: router/gossip_test.go (hunk 2 above)
Comment (dpw) [inline router/gossip_test.go]: I'm just going to delete the comment - `NewTestRouter` seems to cover it.

File: router/router.go (hunk 1 above)
Comment (dpw) [inline router/router.go]: Done

**Pull request merged**